File: generators/chaincode/templates/java/src/main/java/org/example/Chaincode.java
/*
 * <%= spdxAndLicense // SPDX-License-Identifier: Apache-2.0 %>
 */

package org.example;

import java.util.List;

import org.hyperledger.fabric.shim.ChaincodeBase;
import org.hyperledger.fabric.shim.ChaincodeStub;

public class Chaincode extends ChaincodeBase {

    @Override
    public Response init(ChaincodeStub stub) {
        String fcn = stub.getFunction();
        List<String> params = stub.getParameters();
        System.out.printf("init() %s %s\n", fcn, params.toArray());
        return ChaincodeBase.newSuccessResponse();
    }

    @Override
    public Response invoke(ChaincodeStub stub) {
        String fcn = stub.getFunction();
        List<String> params = stub.getParameters();
        System.out.printf("invoke() %s %s\n", fcn, params.toArray());
        return ChaincodeBase.newSuccessResponse();
    }

}

File: generators/chaincode/templates/kotlin/src/main/kotlin/org/example/Chaincode.kt
/*
 * <%= spdxAndLicense // SPDX-License-Identifier: Apache-2.0 %>
 */

package org.example

import org.hyperledger.fabric.shim.Chaincode.Response
import org.hyperledger.fabric.shim.ChaincodeBase
import org.hyperledger.fabric.shim.ChaincodeStub

class Chaincode : ChaincodeBase() {

    override fun init(stub: ChaincodeStub): Response {
        val fcn = stub.getFunction()
        val params = stub.getParameters()
        println("init() ${fcn} ${params}")
        return ChaincodeBase.newSuccessResponse()
    }

    override fun invoke(stub: ChaincodeStub): Response {
        val fcn = stub.getFunction()
        val params = stub.getParameters()
        println("invoke() ${fcn} ${params}")
        return ChaincodeBase.newSuccessResponse()
    }

}
